dagshub/data_engine/model/query.py

```diff
@@ -145,6 +155,10 @@ def compose(self, op: str, other: Optional[Union[str, int, float, "DatasourceQue
             composite_tree.paste(root_node.identifier, self._operand_tree)
             composite_tree.paste(root_node.identifier, other._operand_tree)
             self._operand_tree = composite_tree
+            if other.query_as_of is not None:
+                self.query_as_of = other.query_as_of
+            if other.selects is not None:
+                self.selects = other.selects
```

Comment:
Unclear to me why you'd want to copy this around to the internal nodes, as noted earlier.

Reply:
the new query completely replaces the old one. If you compose a couple queries but don't carry it over, you'll lose the select/as of

Example:
```python
q1 = (ds["data"] == "5").as_of(t1)
q2 = (ds["other_data"] > 5).select("data", "other_data")
q3 = q1 & q2
```
you'd probably expect q3 to persist the select adn as_of, but if we don't carry it over, we'll only get the info from the left-side query, so the as_of would be there, but nto select

Reply:
edge case with multiple selects/as_ofs, don't know how to deal with that one